docs/dev/ERRORS_AND_LOGGING.md

```diff
@@ -0,0 +1,31 @@
+# Errors
+
+* Errors should have a custom variable if they are being used to control execution. Otherwise, create the error at the time of use. 
+
+```
+var ErrMyAwesomeError = errors.New("awesome error")
+err := someFailingFunction()
+if errors.Is(err, ErrMyAwesomeError) {
+  // React to the error
+}
+```
+
+If creating an error this way, it is recommended to add it to an `errors.go` file in the package that is throwing the error.
+
+* Taken from [Working with Errors in Go 1.13](https://go.dev/blog/go1.13-errors):
+> Wrap an error to expose it to callers. Do not wrap an error when doing so would expose implementation details.
+
+When one does wrap an error, it should be wrapped as such:
+```
+fmt.Errorf("my additional context: %w", err)
+```
+
+# Logging
+
+Logging of errors should be limited to only the upper-most layers. Otherwise, wrapped errors could be logged multiple times, causing clutter and limiting the usefulness of the logs.
+
+If an error is logged, and then returned, one should either wrap the current error
+with more context, and return that entire error, or log the wrapped error and then
+return only the wrapper, without the current error.
+
+One notable exception to this is the Builder in the pyxis package. Errors are logged but not returned. Since the errors are not wrapped or propagated up, it makes sense to log them.
```

Comment:
Bit of a far fetched question - do we want to enforce this somehow in GH Actions?

Reply:
Maybe? Seems like a bit of a stretch though. Automated that is. I think to start we as human reviewers have to be more diligent and pay attention to the errors being bandied about.